### PR TITLE
feat : add docs on lst and restaking

### PR DESCRIPTION
**Pull request by 1xstj**
## Summary of changes

_Provide a detailed description of proposed changes._

-

### Proposed area of change

_Put an `x` in the boxes that apply._

- [ ] `Overview`
- [ ] `Anchor System`
- [ ] `dApps`
- [ ] `Tangle`
- [ ] `Protocols`
- [ ] `Relayers`

### Reference issue to close (if applicable)

_Specify any issues that can be closed from these changes (e.g. Closes #233)._

- Closes

---

### Code Checklist

- [ ] I have tested that prove my changes are working as intended
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: pages/restake/assets.mdx [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 ## Supported Assets
 
-At this time, only the native token TNT is supported for restaking participation.
+Supported assets include all "liquid" assets on Tangle Network, including LstTNT. The network does not limit or restrict the assets that can be used for restaking. But assets are
+preffered by adding them to a reward pool.
```
Comment (drewstone) [inline pages/restake/assets.mdx:6]:
```suggestion
preferred by adding them to a reward pool.
```

File: pages/restake/assets.mdx [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
 ## Supported Assets
 
-At this time, only the native token TNT is supported for restaking participation.
+Supported assets include all "liquid" assets on Tangle Network, including LstTNT. The network does not limit or restrict the assets that can be used for restaking. But assets are
```
Comment (drewstone) [inline pages/restake/assets.mdx:5]:
```suggestion
Supported assets include all "liquid" assets on Tangle Network, including liquid staked TNT tokens. The network does not limit or restrict the assets that can be used for restaking. But assets are
```

File: pages/restake/lst-concepts.mdx [hunk 3]
```diff
@@ -1,41 +1,53 @@
...
 
-### Validator and Operator Participation
+## Key Components of Liquid Staking
 
-Validators and operators can stake on their own operations to receive liquid assets. These assets can be used in other DeFi or restaking protocols, providing new yield opportunities.
+### 1. **Pools and Roles**
 
-### Customizability and Control
+- **Pools** are the fundamental unit of staking on Tangle. Pools are managed by specific roles:
+  - **Root**: The administrator who has full control over the pool and can update roles.
```
Comment (drewstone) [inline pages/restake/lst-concepts.mdx:16]: Are these spaces meant to be here?

File: pages/restake/lst-concepts.mdx [hunk 4]
```diff
@@ -1,41 +1,53 @@
...
+### 4. **Withdrawals**
 
-### Liquidity Challenges
+- **Withdraw Unbonded**: Once tokens are unbonded, users can withdraw their funds using the `withdraw_unbonded` function. This function can also be used permissionlessly if certain conditions are met (e.g., the pool is destroying, or the caller is the depositor and the only member left).
 
-Operation-specific LSTs may face liquidity issues in the DeFi ecosystem. Tangle addresses this by encouraging integrations that support trading between LSTs from the same underlying protocol.
+### 5. **Pool Management**
 
-### Security and Slashing
+- **Create Pools**: Users can create new delegation pools with the `create` function. A pool requires an initial deposit, and the creator can set the roles for root, nominator, and bouncer. Pools can also be created with a previously used pool ID using the `create_with_pool_id` function.
```
Comment (drewstone) [inline pages/restake/lst-concepts.mdx:37]: What is a bouncer? Should be defined, no?

File: pages/restake/lst-concepts.mdx [hunk 5]
```diff
@@ -1,41 +1,53 @@
...
+- **Pool State Management**: The state of a pool can be modified using the `set_state` function. Pools can have various states, such as open, blocked, or destroying. Once a pool enters the destroying state, it cannot change its state again.
 
-LSTs serve as collateral for outsourced security and slashing. This ensures network security while enabling innovative staking and restaking solutions.
+### 6. **Rewards and Commission**
 
-### Multi-Chain Support
+- **Commission**: Pools can set a commission percentage using the `set_commission` function. This determines the amount of rewards that are paid to the pool's commission account.
+- **Claiming Commission**: Pending commission can be claimed using the `claim_commission` function, which adds the claimed amount to the total claimed commission and resets pending commission to zero.
+- **Maximum Commission**: The maximum allowable commission can be set with the `set_commission_max` function, ensuring that future commissions are bound by this limit.
+- **Commission Change Rate**: The rate at which commission changes can be controlled using the `set_commission_change_rate` function.
```
Comment (drewstone) [inline pages/restake/lst-concepts.mdx:45]: Not sure I follow why this exists? Can we provide examples too? Does this mean there's a value that defines how much the commission can change from 1 value to another? How is it represented and why is that needed?

File: pages/restake/lst-concepts.mdx [hunk 6]
```diff
@@ -1,41 +1,53 @@
...
 
-Tangle's liquid staking protocol supports multiple blockchains, including Ethereum and Polkadot, allowing users to stake assets across different networks.
+### 7. **Adjusting Pool Deposits**
 
-### Governance and Community Involvement
+- Pools must maintain an existential deposit (ED) in the reward account to ensure proper functionality. The `adjust_pool_deposit` function allows users to either top up the deficit or withdraw any excess deposit from the pool.
```
Comment (drewstone) [inline pages/restake/lst-concepts.mdx:49]: When a pool is created do we deposit the ED into it? When the pool's fees are claimed do we ensure we leave enough to maintain its ED?

**Review by drewstone — CHANGES_REQUESTED** [review]
Looks great @1xstj some comments and things worth adding. I would like to understand an example of when the root pool creator would set a different nominator and a different bouncer. Where did these roles originate from and what is motivation? Separation? Or what? It would be nice to know how it _can_ be used and why.

File: pages/restake/lst-concepts.mdx (hunk 5 above)
Comment (1xstj) [reply pages/restake/lst-concepts.mdx:45]: yes the goal is to prevent a pool creator to advertise 1% commission and after getting deposits to set 10%, this is from the nomination pools pallet, once the upper bound is set, it can only go lower not higher, from docs

```
/// Once a change rate configuration has been set, only more restrictive values can be set
	/// thereafter. These restrictions translate to increased `min_delay` values and decreased
	/// `max_increase` values.``

File: pages/restake/lst-concepts.mdx [hunk 7]
```diff
@@ -1,41 +1,53 @@
...
 
-Governance systems within Tangle can participate in liquid staking, enabling community-driven decisions and improvements, ensuring the protocol benefits all stakeholders.
+### 8. **Claim Permissions**
+
+- The ability to claim pending commission is controlled by the `set_commission_claim_permission` function, which determines who can claim the pool's pending commission.
```
[the remark this answers is not in the document]
Comment (1xstj) [reply pages/restake/lst-concepts.mdx:53]: yes its really for the pool creator to delegate it to another account, the high level goal here is to allow proxy accounts to manage pools while the funds are in a cold wallet, not really necessary but came from nomination pools so didnt remove it

**Review by monaiuu — COMMENTED** [review]
We're currently using the term 'vault' in our UI design, similar to systems like Tenderize.me and Symbiotic. However, understanding that Polkadot.JS uses the term 'pool' for similar functionality and to ensure consistency / avoid user confusion:

- Should we maintain 'vault' in our UI and add clarification in our documentation that this corresponds to 'pools' in Polkadot.JS?

- Or should we align our terminology with Polkadot.JS and change all mentions of 'vault' to 'pool' in our UI for both liquid staking / restaking?

cc @AtelyPham @yurixander

**Comment by AtelyPham** [comment]
> We're currently using the term 'vault' in our UI design, similar to systems like Tenderize.me and Symbiotic. However, understanding that Polkadot.JS uses the term 'pool' for similar functionality and to ensure consistency / avoid user confusion:
> 
> * Should we maintain 'vault' in our UI and add clarification in our documentation that this corresponds to 'pools' in Polkadot.JS?
> * Or should we align our terminology with Polkadot.JS and change all mentions of 'vault' to 'pool' in our UI for both liquid staking / restaking?
> 
> cc @AtelyPham @yurixander

In the context of restaking, we will use "vault" instead of "pool."

**Comment by drewstone** [comment]
Ya worth the-emphasizing @monaiuu that we use the word pool when describing things related to staking and liquid staking (because they're built on pool semantics) but we use the term vaults when we are discussing things related to restaking (because the vault secures services, a la bank vault, etc.). Definitely make sure you establish this semantic difference in your vocabulary. We will be using it moving forward.
